src/main/java/com/commercetools/emailprocessor/email/EmailProcessor.java

```diff
@@ -0,0 +1,134 @@
+package com.commercetools.emailprocessor.email;
+
+
+import com.commercetools.emailprocessor.model.Statistics;
+import com.commercetools.emailprocessor.model.TenantConfiguration;
+import com.fasterxml.jackson.databind.JsonNode;
+import io.sphere.sdk.client.SphereClient;
+import io.sphere.sdk.customobjects.CustomObject;
+import io.sphere.sdk.customobjects.queries.CustomObjectQuery;
+import org.apache.commons.codec.binary.Base64;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.http.NameValuePair;
+import org.apache.http.client.entity.UrlEncodedFormEntity;
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.http.client.methods.HttpPost;
+import org.apache.http.impl.client.HttpClients;
+import org.apache.http.message.BasicNameValuePair;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.crypto.Cipher;
+import javax.crypto.spec.SecretKeySpec;
+import java.nio.charset.Charset;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.CompletionStage;
+
+public class EmailProcessor {
+
+
+    public static final String CONTAINER_ID = "unprocessedEmail";
+    public static final String EMAIL_PROPERTY_STATUS = "status";
+    public static final String EMAIL_STATUS_PENDING = "pending";
+    public static final String EMAIL_STATUS_ERROR = "error";
+    static final String PARAM_EMAIL_ID = "emailid";
+    static final String PARAM_TENANT_ID = "tenantid";
+    private static final String ENCRYPTION_ALGORITHM = "Blowfish";
+    private static final Logger LOG = LoggerFactory.getLogger(EmailProcessor.class);
+
+    public EmailProcessor() {
+    }
+
+    /**
+     * Iterate through all Email objects and triggers the webhook for each  pending email object.
+     *
+     * @param tenantConfiguration Configuration of a tenant
+     * @return Statics of the sended emails
+     */
+
+    public CompletionStage<Statistics> processEmails(final TenantConfiguration tenantConfiguration) {
+        SphereClient client = tenantConfiguration.getSphereClient();
```

Comment:
what if tenant configuration is passed `null`? you will hit NPE here

Reply:
a4d5d9e